Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,408 @@
+
+      guard let index = protos.firstIndex(where: { $0.name == fileName }) else {
+        return XCTFail(
+          """
+          Could not find the file descriptor proto of \(fileName)
+          in the original file descriptor protos list.
+          """
+        )
+      }
+
+      let originalProto = protos[index]
+      XCTAssertEqual(originalProto.name, fileName)
+      XCTAssertEqual(try originalProto.serializedData(), serializedFiledescriptorData)
+      XCTAssertEqual(originalProto.dependency, dependencyFileNames)
+
+      protos.remove(at: index)
+    }
+    XCTAssert(protos.isEmpty)
+  }
+
+  func testReflectionServiceServicesNames() throws {
+    let protos = self.makeProtosWithDependencies()
+    let servicesNames = self.getServicesNamesFromProtos(protos: protos).sorted()
+    let registry = try ReflectionServiceData(fileDescriptors: protos)
+    let registryServices = registry.serviceNames.sorted()
+    XCTAssertEqual(registryServices.count, servicesNames.count)
+    for index in 0 ... servicesNames.count - 1 {
+      XCTAssertEqual(registryServices[index], servicesNames[index])
+    }
+  }
+
+  func testSerialisedFileDescriptorProtosForDependenciesOfFile() throws {
+    var protos = self.makeProtosWithDependencies()
+    let registry = try ReflectionServiceData(fileDescriptors: protos)
+    let serializedFileDescriptorProtos =
+      try registry
+      .serialisedFileDescriptorProtosForDependenciesOfFile(named: "bar1.proto")
+    let fileDescriptorProtos = try serializedFileDescriptorProtos.map {
+      try Google_Protobuf_FileDescriptorProto(serializedData: $0)
+    }
+    // Tests that the functions returns all the transitive dependencies, with their services and
+    // methods, together with the initial proto, as serialized data.
+    XCTAssertEqual(fileDescriptorProtos.count, 4)
+    for fileDescriptorProto in fileDescriptorProtos {
+      guard let protoIndex = protos.firstIndex(of: fileDescriptorProto) else {
+        return XCTFail(
+          """
+          Could not find the file descriptor proto of \(fileDescriptorProto.name)
+          in the original file descriptor protos list.
+          """
+        )
+      }
+      for service in fileDescriptorProto.service {
+        guard let serviceIndex = protos[protoIndex].service.firstIndex(of: service) else {
+          return XCTFail(
+            """
+            Could not find the \(service.name) in the service
+            list of the \(fileDescriptorProto.name) file descriptor proto.
+            """
+          )
+        }
+        let originalMethods = protos[protoIndex].service[serviceIndex].method
+        for method in service.method {
+          XCTAssert(originalMethods.contains(method))
+        }
+        for messageType in fileDescriptorProto.messageType {
+          XCTAssert(protos[protoIndex].messageType.contains(messageType))
+        }
+      }
+      protos.removeAll { $0 == fileDescriptorProto }
```

Comment:
Newlines help readability, it usually helps to put one between each "block" of code

```suggestion
      guard let protoIndex = protos.firstIndex(of: fileDescriptorProto) else {
        return XCTFail(
          """
          Could not find the file descriptor proto of \(fileDescriptorProto.name)
          in the original file descriptor protos list.
          """
        )
      }
      
      for service in fileDescriptorProto.service {
        guard let serviceIndex = protos[protoIndex].service.firstIndex(of: service) else {
          return XCTFail(
            """
            Could not find the \(service.name) in the service
            list of the \(fileDescriptorProto.name) file descriptor proto.
            """
          )
        }
        
        let originalMethods = protos[protoIndex].service[serviceIndex].method
        for method in service.method {
          XCTAssert(originalMethods.contains(method))
        }
        
        for messageType in fileDescriptorProto.messageType {
          XCTAssert(protos[protoIndex].messageType.contains(messageType))
        }
      }
      
      protos.removeAll { $0 == fileDescriptorProto }
```